src/main/java/uk/gov/hmcts/reform/sscs/transform/deserialize/SubmitYourAppealToCcdCaseDataDeserializer.java

```diff
@@ -99,9 +102,32 @@ public static SscsCaseData convertSyaToCcdCaseData(SyaCaseWrapper syaCaseWrapper
             .benefitCode(benefitCode)
             .issueCode(issueCode)
             .caseCode(caseCode)
+            .dwpRegionalCentre(getDwpRegionalCenterGivenDwpIssuingOffice(appeal.getBenefitType().getCode(),
+                appeal.getMrnDetails().getDwpIssuingOffice()))
             .build();
     }
 
+    private static String getDwpRegionalCenterGivenDwpIssuingOffice(String benefitTypeCode, String dwpIssuingOffice) {
+        if ("UC".equals(benefitTypeCode)) {
+            DwpAddressLookupService dwpAddressLookupService = new DwpAddressLookupService();
+            Optional<OfficeMapping> officeMapping = dwpAddressLookupService.getDwpMappingByOffice(
+                benefitTypeCode, dwpIssuingOffice);
+            return officeMapping.map(mapping -> mapping.getMapping().getCcd()).orElse(null);
+        }
+        if (dwpIssuingOffice != null && benefitTypeCode != null) {
+            DwpAddressLookupService dwpAddressLookupService = new DwpAddressLookupService();
+            Optional<OfficeMapping> officeMapping = dwpAddressLookupService.getDwpMappingByOffice(
+                benefitTypeCode, dwpIssuingOffice);
+            if ("ESA".equals(benefitTypeCode)) {
+                return officeMapping.map(mapping -> mapping.getMapping().getCcd()).orElse(null);
```

Comment:
Why is ESA and UC looking at `getCcd()` field but PIP looking at `getDwpRegionCentre()`? Shouldn't they all be `getDwpRegionCentre()`?

Reply:
there is no dwpRegionalCentre for UC and ESA. Then mapping.ccd data meet the ticket requirements so I used that.